packages/backend/src/tokens/request.ts

```diff
@@ -148,10 +150,33 @@ function assertProxyUrlOrDomain(proxyUrlOrDomain: string | undefined) {
 }
 
 export async function authenticateRequest(options: AuthenticateRequestOptions): Promise<RequestState> {
-  options.frontendApi = parsePublishableKey(options.publishableKey)?.frontendApi || options.frontendApi || '';
-  options.apiUrl = options.apiUrl || API_URL;
-  options.apiVersion = options.apiVersion || API_VERSION;
-  options.headerToken = options.headerToken?.replace('Bearer ', '');
+  const { request: isomorphicRequest } = options;
+  const isomorphicRequestCookies = isomorphicRequest ? parseIsomorphicRequestCookies(isomorphicRequest) : undefined;
+  const isomorphicRequestSearchParams = isomorphicRequest?.url
+    ? new URL(isomorphicRequest.url)?.searchParams
+    : undefined;
+
+  const getHeaderFromIsomorphicReq = (key: string) => isomorphicRequest?.headers?.get(key) || undefined;
```

Comment:
🔧 I'd move all these utilities under the IsomorphicRequest utility object. For example:

```
function buildRequest(options...) {
  return {
    getHeader(),
    getCookie(),
    getBody(),
}
```

Reply:
Done!